language/move-lang/tests/move_check_testsuite.rs

```diff
@@ -50,7 +50,11 @@ fn move_check_testsuite(path: &Path) -> datatest_stable::Result<()> {
     let exp_path = path.with_extension(EXP_EXT);
     let out_path = path.with_extension(OUT_EXT);
 
-    let (files, errors) = move_check_no_report(&targets, &deps, sender)?;
+    let (files, units_or_errors) = move_compile_no_report(&targets, &deps, sender)?;
+    let errors = match units_or_errors {
```

Comment:
`.map(|units| move_lang::  ...)`?

Reply:
`map` doesn't really work here. I would need something like `(Result<T, E>, FnOnce(T) ->  E) -> E`, and I don't think that exists. `map_or_else`  is sort of close, but you would need the identity for the Errors case, and I don't think that code would be any cleaner than the `match` unfortunately